Documentation/Design/Lambdas/Lambda Declarations.md

```diff
@@ -0,0 +1,150 @@
+# Lambda Declarations
+
+GLS will support declaring lambda functions. In some languages, in order to declare a function that takes a lambda function as a parameter, the lambda must first be declared. This typically involves declaring the return type as well as the parameter names and types, which allow for compile-time type checking when the lambda is used.
+
+## Commands
+
+### `lambda declare`
+`lambda declare : typeName functionName returnType *`[parameterName parameterType ...]`*
+
+## Usage:
+
+```gls
+lambda declare : overMinimum check boolean age int
```

Comment:
I meant it to be used in a bar sort of example, but the over21 -> overMinimum change made that less obvious. I shall make it more innocent.